helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/filter/range/FilterRangeConnectionSearcher.kt

```diff
@@ -0,0 +1,73 @@
+@file:Suppress("FunctionName")
+
+package com.algolia.instantsearch.helper.filter.range
+
+import com.algolia.instantsearch.core.connection.Connection
+import com.algolia.instantsearch.core.connection.ConnectionImpl
+import com.algolia.instantsearch.core.number.range.Range
+import com.algolia.instantsearch.helper.filter.range.internal.mapper
+import com.algolia.instantsearch.helper.searcher.SearcherSingleIndex
+import com.algolia.search.model.Attribute
+import com.algolia.search.model.response.ResponseSearch
+import com.algolia.search.model.search.FacetStats
+import com.algolia.search.model.search.Query
+import kotlin.reflect.KClass
+
+internal class FilterRangeConnectionSearcherImpl<T>(
+    private val filterRangeViewModel: FilterRangeViewModel<T>,
+    private val searcher: SearcherSingleIndex,
+    private val attribute: Attribute,
+    private val mapper: (Number) -> T
+) : ConnectionImpl() where T : Number, T : Comparable<T> {
+
+    val subscription: (ResponseSearch?) -> Unit = { response ->
+        filterRangeViewModel.computeBoundsFromFacetStats(attribute, response?.facetStats, mapper)
+    }
+
+    private fun <T> FilterRangeViewModel<T>.computeBoundsFromFacetStats(
+        attribute: Attribute,
+        facetStats: Map<Attribute, FacetStats>?,
+        mapper: (Number) -> T
+    ) where T : Number, T : Comparable<T> {
+
+        val facetStatsOfAttribute = facetStats?.get(attribute)
+        if (facetStatsOfAttribute == null) {
+            bounds.value = null
+            return
+        }
+
+        val min = mapper(facetStatsOfAttribute.min)
+        val max = mapper(facetStatsOfAttribute.max)
+
+        bounds.value = Range(min, max)
+    }
+
+    override fun connect() {
+        super.connect()
+        searcher.query.updateQueryFacets(attribute)
+        searcher.response.subscribePast(subscription)
+    }
+
+    private fun Query.updateQueryFacets(attribute: Attribute) {
+        facets = if (facets?.contains(attribute) == false) {
+            val current = facets?.toMutableSet() ?: mutableSetOf()
+            current + attribute
+        } else {
+            setOf(attribute)
+        }
+    }
+
+    override fun disconnect() {
+        super.disconnect()
+        searcher.response.unsubscribe(subscription)
+    }
+}
+
+public fun <T> FilterRangeConnectionSearcher(
```

Comment:
After reading the internal business logic, I understood what `clazz` is for, but I would say we need to document it and list all the available classes, because it is unclear. 

Also, I'm not sure if it's possible to do some sort of `enum` that has all the valid classes, so that we get some sort of compile safety?

Reply:
This function, normally, should not be used directly except if the user knows what he is doing. The overload with the `mapper` parameter is exactly for this: to give the possibility for a custom behavior.  I agree that we should document this.

The `KClass<T>` is inferred automatically using the main entry inline functions (`connectSearcher` extension functions), unfortunately, we cannot make this builder functions internal since a public `inline` function cannot "expose" `internal` function/classes.

Reply:
By the way, all this is only to bypass the limitation of generics type erasure, especially at the class level, without introducing a breaking change or reflection.

Reply:
Got it 👍